web/src/narrow_banner.ts

```diff
@@ -127,6 +129,51 @@ export function pick_empty_narrow_banner(): NarrowBannerData {
 
     const current_filter = narrow_state.filter();
 
+    const stream_id = current_filter?._sub?.stream_id;
+
+    const muted_banner = {
+        title: $t({
+            defaultMessage:
+                "This feed is empty, because you have muted all the topics in this channel.",
+        }),
+        html: page_params.is_spectator
+            ? ""
+            : $t_html(
+                  {
+                      defaultMessage: "Why not <z-link>start the conversation</z-link>?",
+                  },
+                  {
+                      "z-link": (content_html) =>
+                          `<a href="#" class="empty_feed_compose_stream">${content_html.join(
+                              "",
+                          )}</a>`,
+                  },
+              ),
+    };
+
+    const sub_dict = all_user_topics.get(stream_id!);
+    if (sub_dict) {
+        let all_topics_muted = true;
+        let has_messages = false;
+
+        for (const [topic, value] of sub_dict) {
+            const topic_has_messages = get_messages_in_topic(stream_id!, topic).length > 0;
```

Comment:
This is extremely inefficient; you're asking for all the messages in every topic in this channel individually.